torchvision/prototype/ops/_utils.py

```diff
@@ -0,0 +1,119 @@
+import copy
+import operator
+import warnings
+from typing import Callable, Tuple, Union
+
+import torch
+from torch import fx
+from torchvision.models.feature_extraction import LeafModuleAwareTracer
+
+
+_MODULE_NAME = "_regularized_shotrcut"
+
+
+class RegularizedShortcut(torch.nn.Module):
+    def __init__(self, regularizer_layer: Callable[..., torch.nn.Module]):
+        super().__init__()
+        self._regularizer = regularizer_layer()
+
+    def forward(self, input, result):
+        return input + self._regularizer(result)
+
+
+def add_regularized_shortcut(
+    model: torch.nn.Module,
+    block_types: Union[type, Tuple[type, ...]],
+    regularizer_layer: Callable[..., torch.nn.Module],
+    inplace: bool = True,
+) -> torch.nn.Module:
+    if not inplace:
+        model = copy.deepcopy(model)
+
+    tracer = fx.Tracer()
+    modifications = {}
+    for name, m in model.named_modules():
+        if isinstance(m, block_types):
+            # Add the Layer directly on submodule prior tracing
+            # workaround due to https://github.com/pytorch/pytorch/issues/66197
+            m.add_module(_MODULE_NAME, RegularizedShortcut(regularizer_layer))
```

Comment:
Ideally I wanted to create the Layer on the fly and attach it directly on the graph but the https://github.com/pytorch/pytorch/issues/66197 issue prohibits me from doing this. Here I attach it to the module just before tracing it as a workaround. Any concerns?

It will be removed once the issue is fixed.